specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -6888,6 +7358,280 @@
     }
   },
   "definitions": {
+    "RestorePointCollectionProperties": {
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/SubResource",
+          "description": "Reference Id of the VM for which the restore points have to be created."
+        },
+        "restorePoints": {
```

Comment:
>restorePoints [](start = 9, length = 13)

Is this array the same as the nested /restorePoints resource type?

Reply:
sorry, could you please clarify the question? Does the / have a special significance?

Customers can query restorePointCollection to get a list of all restore points that the restorePointCollection contains.
The array is a list of restore points.

Customer can also query restore point specifically by doing
GET     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{restorePointName}"

Reply:
Modeling the same information as an array property on the parent resource AND a nested resource type causes problems in Policy, RBAC, Deployments, and ARG.

Is it possible to avoid having restore points in two places? Or replace the restorePoints property on the collection with an array of links to the restore points rather than including the entire restore point content? (Like the `SubResource` pattern in NRP.)

Reply:
My manager (Sameer Motwani) said we (CRP team) had reviewed this pattern with ARM team with few years back and we had gotten permission for this.
Also, we have partners (such as BackupService) that have been using this api for years now. Changing it now will result in breaking changes for them, which we cannot do.
Also, we have other examples of this pattern in the CRP which were also reviewed by ARM team.

This particular Swagger PR for RestorePoints is just adding these apis now to client tools to make it available to more customers, but didn't plan on changing our existing API (especially because they're already being used by customers).